python/lsst/sims/catUtils/utils/SNIaLightCurveGenerator.py

```diff
@@ -0,0 +1,164 @@
+                                for ff, ff_sn in zip(flambda_grid, sn_ff_buffer):
+                                    ff[mask] = np.where(ff_sn > 0.0, ff_sn, 0.0)
+
+                                ss = Sed()
+                                fnu_grid = flambda_grid*bandpass.wavelen* \
+                                           bandpass.wavelen*ss._physParams.nm2m* \
+                                           ss._physParams.ergsetc2jansky/ss._physParams.lightspeed
+
+                                flux_list = \
+                                (fnu_grid*bandpass.phi).sum(axis=1)*(bandpass.wavelen[1]-bandpass.wavelen[0])
+
+                                flux_list = np.array(flux_list)
+                                ss = Sed()
+                                mag_list = ss.magFromFlux(flux_list)
+
+                                acceptable = np.where(np.isfinite(mag_list))
+                                mag_error_list = calcMagError_m5(mag_list[acceptable], bandpass,
+                                                                 m5_active[acceptable], self.phot_params,
+                                                                 gamma=gamma_active[acceptable])
+
+                                if len(acceptable) > 0:
+
+                                    if sn[0] not in self.truth_dict:
+                                        self.truth_dict[sn[0]] = {}
+                                        self.truth_dict[sn[0]]['t0'] = sn_t0
+                                        self.truth_dict[sn[0]]['x1'] = sn_x1
+                                        self.truth_dict[sn[0]]['x0'] = sn_x0
+                                        self.truth_dict[sn[0]]['c'] = sn_c
+                                        self.truth_dict[sn[0]]['z'] = sn[5]
+                                        self.truth_dict[sn[0]]['E(B-V)'] = sn[6]
+
+                                    if sn[0] not in self.mjd_dict:
+                                        self.mjd_dict[sn[0]] = {}
+                                        self.mag_dict[sn[0]] = {}
+                                        self.sig_dict[sn[0]] = {}
+
+                                    if bp_name not in self.mjd_dict[sn[0]]:
+                                        self.mjd_dict[sn[0]][bp_name] = []
+                                        self.mag_dict[sn[0]][bp_name] = []
+                                        self.sig_dict[sn[0]][bp_name] = []
+
+                                for tt, mm, ee in zip(t_active[acceptable], mag_list[acceptable],
+                                                      mag_error_list[0]):
+
+                                    self.mjd_dict[sn[0]][bp_name].append(tt)
+                                    self.mag_dict[sn[0]][bp_name].append(mm)
+                                    self.sig_dict[sn[0]][bp_name].append(ee)
+
+            print("chunk of ", len(chunk), " took ", time.time()-t_start_chunk)
```

Comment:
Does this mean that flux and flux errror  which are the quantities of interest for SN, are not returned?